### PR TITLE
Add Core-compatible format of RecoverableSignature

### PR DESCRIPTION
**Pull request by stevenroose**
Bitcoin Core produces a 65-byte signature when signing messages.  The
recovery id is encoded in the first byte.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -347,6 +347,43 @@ impl RecoverableSignature {
...
+        unsafe {
+            let err = ffi::secp256k1_ecdsa_recoverable_signature_serialize_compact(
+                secp.ctx,
+                ret[1..].as_mut_ptr(),
+                &mut recid,
+                self.as_ptr(),
+            );
+            assert!(err == 1);
+        }
+        ret[0] = (27 + 4 + recid) as u8; // compressed
```
Comment (stevenroose) [inline src/lib.rs]: This is a bit of a problem. It looks like Bitcoin Core sets the compressed flag based on the public key, which we don't have here of course.

**Comment by stevenroose** [comment]
The format is basically defined here: https://github.com/bitcoin/bitcoin/blob/master/src/key.cpp#L241-L254

**Comment by stevenroose** [comment]
Given that serializing these signatures is dependent on the public key, I removed the serialization method and only provided the parsing method. That one should work fine in any case.

**Comment by stevenroose** [comment]
Feel free to ignore it either way, it's not that hard to just have this method ad-hoc where it's needed.

**Comment by apoelstra** [comment]
I'm going to close this -- I don't think we can do enough to be useful within the scope of this library (and I'm unconvinced that we want to do anything useful wrt recoverable signatures, anyway :)).

In particular, to be compatible with sign/verifymessage there is still a base64 conversion that needs to be done on the signatures and an address computation/encoding that needs to be done on the public keys.

**Comment by stevenroose** [comment]
Fair enough :)

**Pull request closed without merging**